Source/trigs.cpp

```diff
@@ -823,108 +823,83 @@ void CheckTrigForce()
 
 void CheckTriggers()
 {
-	int *v0;       // edi
-	int v1;        // esi
-	int v2;        // ecx
-	int v3;        // eax
-	int v4;        // edx
-	signed int v5; // edx
-	int v6;        // eax
-	int v9;        // [esp-4h] [ebp-20h]
-	int x;         // [esp+Ch] [ebp-10h]
-	int y;         // [esp+10h] [ebp-Ch]
-	int v12;       // [esp+14h] [ebp-8h]
-	int error_id;  // [esp+1Bh] [ebp-1h]
+	int x, y, i;
+	BOOL abort;
+	char abortflag;
 
 	if (plr[myplr]._pmode)
 		return;
-	v12 = 0;
-	if (trigflag[4] <= 0)
-		return;
-	v0 = &trigs[0]._tmsg;
-	while (1) {
-		v1 = myplr;
-		v2 = plr[myplr].WorldX;
-		if (v2 != *(v0 - 2))
-			goto LABEL_34;
-		v3 = plr[v1].WorldY;
-		if (v3 != *(v0 - 1))
-			goto LABEL_34;
-		v4 = *v0;
-		if (*v0 == WM_DIABNEXTLVL) {
+
+	for (i = 0; i < trigflag[4]; i++) {
+		if (trigs[i]._tx != plr[myplr].WorldX || trigs[i]._ty != plr[myplr].WorldY) {

```

Comment:
Probably this will fix the diff:
```suggestion
		if (plr[myplr].WorldX != trigs[i]._tx || plr[myplr].WorldY != trigs[i]._ty) {
```